cmd/loadTests.go

```diff
@@ -544,12 +560,17 @@ func userJourneyThread(frameworkMap *sync.Map, threadWaitGroup *sync.WaitGroup,
 					continue
 				}
 				usernamespace := framework.UserNamespace
-				ComponentName := fmt.Sprintf("%s-component", username)
-				ApplicationName := fmt.Sprintf("%s-app", username)
-				DefaultRetryInterval := time.Second * 5
-				DefaultTimeout := time.Minute * 60
-				error := k8swait.Poll(DefaultRetryInterval, DefaultTimeout, func() (done bool, err error) {
-					pipelineRun, err := framework.AsKubeDeveloper.HasController.GetComponentPipelineRun(ComponentName, ApplicationName, usernamespace, "")
+				componentName := componentForUser(username)
+				if componentName == "" {
+					logError(9, fmt.Sprintf("Component not found for username %s", username))
+					increaseBar(pipelinesBar, pipelinesBarMutex)
+					continue
+				}
+				applicationName := fmt.Sprintf("%s-app", username)
+				pipelineRetryInterval := time.Second * 5
+				pipelineTimeout := time.Minute * 60
+				error := k8swait.Poll(pipelineRetryInterval, pipelineTimeout, func() (done bool, err error) {
+					pipelineRun, err := framework.AsKubeDeveloper.HasController.GetComponentPipelineRun(componentName, applicationName, usernamespace, "")
```

Comment:
Component name is now generated by HAS with the use of `ComonentDetectionQuery` and is different from `<username>-component`. Keeping the track of component names per user via thread-safe `userComponentMap` map.